File: backend/app/schemas/dataset.py
from datetime import datetime

from pydantic import BaseModel, Field


class LabelClassConfig(BaseModel):
    name: str
    color: str


class DatasetCreate(BaseModel):
    name: str = Field(..., max_length=255)
    description: str | None = None
    source_type: str = "upload"
    label_config: list[LabelClassConfig] = Field(default_factory=list)


class DatasetUpdate(BaseModel):
    name: str | None = Field(None, max_length=255)
    description: str | None = None
    status: str | None = None
    label_config: list[LabelClassConfig] | None = None


class DatasetOut(BaseModel):
    id: str
    organization_id: str
    project_id: str
    name: str
    description: str | None
    source_type: str
    s3_bucket: str
    s3_prefix: str
    image_count: int
    total_size_bytes: int
    status: str
    label_config: list[LabelClassConfig]
    preprocessing_config: dict | None
    annotation_project_id: str | None
    created_by: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/dataset.ts
export interface LabelClassConfig {
  name: string;
  color: string;
}

export interface Dataset {
  id: string;
  organization_id: string;
  project_id: string;
  name: string;
  description: string | null;
  source_type: string;
  s3_bucket: string;
  s3_prefix: string;
  image_count: number;
  total_size_bytes: number;
  status: string;
  label_config: LabelClassConfig[];
  preprocessing_config: Record<string, unknown> | null;
  annotation_project_id: string | null;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface CreateDatasetRequest {
  name: string;
  description?: string;
  source_type?: string;
  label_config: LabelClassConfig[];
}

export interface UpdateDatasetRequest {
  name?: string;
  description?: string;
  status?: string;
  label_config?: LabelClassConfig[];
}
